### DISCUSSION TITLE
Name tag of the avatar in display becomes half once maximum number of people enter into hubs cloud room

### DISCUSSION THREAD

**Discussion by Suryasgithub7** [mozilla/hubs#5884] [Hubs Cloud Questions]
**Description**
When more than 10 or maximum number of people enter into the room. Name tag of the avatar changes from fully visible to only below half of the name as displayed like attached screenshots.

**To Reproduce**
1. Go to the hubs cloud room
2. Join more than 10 people - Approximately (When maximum number of people enter into their room)
3. After the maximum people entered the name of the avatar will becomes half as looking in image.

**Expected behavior**
Name tag of the avatar should be displayed as same as full name how it is before the maximum people entered..

**Screenshots**

![image](https://user-images.githubusercontent.com/119397630/210948297-705f24e1-aa4b-4986-8c77-dd0e5b757be0.png)

![image](https://user-images.githubusercontent.com/119397630/210948660-b72f6a97-e42f-4ccb-87b5-b07273e3eb34.png)

**File Path：**
hubs>src>components>troika-text.js

**Hardware** 
 - Device: [Desktop]
 - OS: [e.g. Windows]
 - Browser: [e.g. Chrome]

**To be noted**
In JavaScript code the previous font URL is mentioned in English font then later it is changed to Japanese font.
The highlighted code in image consists of font URL for Japanese font.

**Additional context**

Did anybody had faced similar situation, Also if any inputs regarding it to resolve the issue would be much appreciated. Thanks in advance

JavaScript have been used for displaying the name font. Kindly check the attached screenshot of the JavaScript code

**Comment by matthewbcool** [comment]
Converting to a hubs cloud discussion since we can't reproduce this outside of this custom client.

Are we sure that this only occurs when there are over 10 people in the room? If you throttle your internet speed using dev tools, can you reproduce this issue I wonder?

I would guess you may have to make some further customizations maybe to the troika text component directly: https://protectwise.github.io/troika/troika-three-text/

**Reply by Suryasgithub7** [reply]
@matthewbcool 
Thank you for reply.
We have checked the following information, but still do not know the cause of the occurrence.
We would appreciate your advice on how to resolve this issue after reviewing the following. ## Additional things we have confirmed
- When other users update their names, or when other users enter or leave the room, the text is broken. The text is garbled as well as broken text.
- It seems that Troika Text is also used for the open-media-button, and when I create a new room with 100 media buttons, and enter the room with the media buttons set to long Japanese text, the text is broken regardless of the number of other users. The text is corrupted regardless of the number of other users (except zero).
 ![image](https://user-images.githubusercontent.com/105335052/217702951-cdab8ae5-fdde-4f92-9b6c-4f1d5b2cadff.png)
    - However, if there are no other users in the room, no matter how many media buttons are set, the text will not be corrupted.
    - So, The trigger is not the media button, but rather the **update of the name tag.**
- I tried the following font files, but all of them caused text corruption
    - HigashiOme-Gothic-1.3i.ttf
    - NotoSansJP-Black.otf
    - NotoSansTC-Regular.otf
    - OpenSans-Italic.ttf
    - The above font files were applied by changing the default fontUrl in troika-text.js.
- We checked where troika-three-text was imported, but could not determine the cause of the error.
- Once this occurred, other users' name tags and media buttons collapsed at the same time. If other users update their names again, they will not be restored. (Reloading the screen will restore it)
- Not all logged in users experience the text collapse at the same time, but it occurs on a device/browser by device/browser basis. 
## Questions
- Is it an internal problem of troika-three-text or a problem in the part of the Hubs that manages 3d characters, such as name-tag.js?
- Is there a problem with the character mesh creation that is commonly done? Or is it something else?
- Is the cause due to the complexity of the character mesh, or is it that the font sometimes does not load correctly? Or is the character code the cause?
